activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -361,6 +361,16 @@ def test_unscoped_with_named_scope_should_not_have_default_scope
     assert_equal 10, DeveloperCalledJamis.unscoped { DeveloperCalledJamis.poor }.length
   end
 
+  def test_default_scope_with_joins
+    assert_equal 4,  Comment.joins(:special_post_with_default_scope).length
+    assert_equal 11, Comment.joins(:post).length
```

Comment:
That seemed a magick number. I wrote code to count them. 16faa5f85fdee62723bab8f428d7a1455e2bd00f